rfcs/enum-merging.md

```diff
@@ -0,0 +1,291 @@
+- **Output Enums**: A subgraph returns an enum value not defined in the gateway's schema, causing serialization errors at the gateway.
+
+**Examples:**
+
+- **Input Error**: A client attempts to create an order with `shippingMethod: EXPRESS`, but Subgraph A, which handles order creation, does not recognize `EXPRESS`, leading to a runtime error in the subgraph.
+  
+  ```graphql
+  mutation {
+    createOrder(input: { shippingMethod: EXPRESS }) {
+      order {
+        id
+        shippingMethod
+      }
+    }
+  }
+  ```
+
+- **Output Error**: Subgraph B returns an order with `shippingMethod: EXPRESS`, but if the gateway's enum does not include `EXPRESS`, the gateway cannot serialize the response, resulting in a runtime error.
+
+### 2. Errors Not Surfacing Early
+
+When the there are no composition-time errors, developers may not be aware of inconsistencies until they manifest as runtime failures in production. Preferably, errors should be caught during schema composition to provide immediate feedback to developers.
+
+### 3. Difficulty in Schema Evolution
+
+Exposing a superset or intersection makes evolving the schema challenging:
+
+- **Removing Values**: Removing an enum value is complicated if subgraphs still rely on it.
+- **Adding Values**: Introducing a new enum value requires coordination among all subgraphs to prevent inconsistencies.
+
+## Proposed Solutions
+
+To address these problems, we consider three options for handling enum merging. Each option is analyzed with integrated pros, cons, and examples.
+
+### Option 1: Strict Equality with Inaccessible Values (Recommended)
+
+#### Approach
+
+- **Enforce Strict Equality**: Enum definitions must be identical across subgraphs.
+- **Use `@inaccessible` Directive**: Subgraph-specific enum values are annotated with `@inaccessible` to hide them from the gateway schema.
+
+#### Implementation
+
+**Subgraph A**
+
+```graphql
+enum ShippingMethod {
+  STANDARD
+  ECONOMY
+  INTERNATIONAL @inaccessible
+}
+```
+
+**Subgraph B**
+
+```graphql
+enum ShippingMethod {
+  STANDARD 
+  ECONOMY 
+  EXPRESS @inaccessible
+  INTERNATIONAL
+}
+```
+
+**Gateway**
+
+```graphql
+enum ShippingMethod {
+  STANDARD
+  ECONOMY
+}
+```
+
+#### Pros and Cons
+
+**Pros:**
+
+- **Prevents Runtime Errors**: By ensuring that only recognized enum values are exposed, runtime errors are minimized.
+- **Composition-Time Validation**: Differences in enum definitions are caught during schema composition.
+- **Explicit Control**: Developers explicitly manage enum exposure, increasing schema clarity, with the hope that they will not return the inaccessible values until they are exposed.
+- **Supports Evolvability**: A new enum value can be added by flagging it as `@inaccessible` in a subgraph which hides it from the gateway schema. The other subgraph can then expose it when ready. The flag can be removed once all subgraphs are updated. 
```

Comment:
If we have one subgraph, the first subgraph can add `@inaccessible` but then the other two can add the enum with out marking it as `@inaccessible`. That is by design of the current `@inaccessible` directive today with fields but should we make that more explicit with enums?

Maybe we should make it so everyone has to add `@inaccessible`? But then everyone has to also remove it to release and that is more work and time....